src/conn_handler.c

```diff
@@ -288,22 +288,42 @@ static void* flush_thread(void *arg) {
     // Cast the args
     metrics *m = arg;
 
+    struct timeval tv_start, tv_end, tv_diff;
+
     // Get the current time
-    struct timeval tv;
-    gettimeofday(&tv, NULL);
+    gettimeofday(&tv_start, NULL);
 
     // Determine which callback to use
     stream_callback cb = (GLOBAL_CONFIG->binary_stream)? stream_formatter_bin: stream_formatter;
 
     // Stream the records
-    int res = stream_to_command(m, &tv, cb, GLOBAL_CONFIG->stream_cmd);
+    int res = stream_to_command(m, &tv_start, cb, GLOBAL_CONFIG->stream_cmd);
     if (res != 0) {
         syslog(LOG_WARNING, "Streaming command exited with status %d", res);
     }
 
     // Cleanup
     destroy_metrics(m);
     free(m);
+
+    // Get time passed so that we can report on
+    // how long it took us to flush the metrics
+    gettimeofday(&tv_end, NULL);
+    int tv_diff_usec = ((tv_end.tv_sec * 1000000) + tv_end.tv_usec) - ((tv_start.tv_sec * 1000000) + tv_start.tv_usec);
+
+    syslog(LOG_INFO, "Flushed metrics [%i usecs]", tv_diff_usec);
+
+    // By this point, GLOBAL_METRICS has subbed out, and so we can add
+    // our new metrics into the new set.
+    if (GLOBAL_CONFIG->metrics_prefix) {
+        char* metric_key = malloc(strlen(GLOBAL_CONFIG->metrics_prefix)+100);
+        metric_key[0] = '\0';
+        strcat(metric_key, GLOBAL_CONFIG->metrics_prefix);
+        strcat(metric_key, ".flush.time_ms");
+        metrics_add_sample(GLOBAL_METRICS, TIMER, metric_key, tv_diff_usec/1000, 1.0, GLOBAL_CONFIG->metrics_prefix);
+    }
```

Comment:
Remember to call `free` (I think we can do it after metrics_add_sample since the string has been used by that point)

Reply:
true that!